### ISSUE TITLE
[Bug] Do not override llama pad token if exists

### ISSUE THREAD

**Issue by NanoCode012** [axolotl-ai-cloud/axolotl#385]
https://github.com/OpenAccess-AI-Collective/axolotl/blob/919246fbc1f6d925e792703d8fe989d05e200029/src/axolotl/utils/models.py#L57-L61

```py
        if not (tokenizer.pad_token or cfg.special_tokens.get("pad_token")):
            LOG.warning(
                "LlamaTokenizer does not have a pad token. Setting to LLAMA_DEFAULT_PAD_TOKEN"
            )
            tokenizer.pad_token = LLAMA_DEFAULT_PAD_TOKEN
```

**Comment by winglian** [comment]
@NanoCode012 is this still something we need to fix?

**Comment by NanoCode012** [comment]
This is now fixed with #475 and #477
